### PR TITLE
Adding animation to referral link button

### PR DESCRIPTION
**Pull request by axelbacklund**
Nice spinning animation to the referral link

### CODE REVIEW

File: Src/Generic/Viewables/LoadableButton.swift [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            make.height.equalToSuperview().multipliedBy(0.7)
+            make.center.equalToSuperview()
+        }
+        
+        bag += buttonView.didLayoutSignal.onValue { _ in
+            buttonView.titleLabel?.frame.size.width = buttonView.titleLabel?.intrinsicContentSize.width ?? 0
+        }
+        
+        func setLoadingState(isLoading: Bool, animate: Bool) {
+            buttonView.snp.updateConstraints({ (make) in
```
Comment (Iamsamwhoami) [inline Src/Generic/Viewables/LoadableButton.swift]: remove parenthesis around make and the whole closure.

File: Src/Generic/Viewables/LoadableButton.swift [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            
+            if animate {
+                let labelDelay = isLoading ? 0 : 0.25
+                let layoutDelay = isLoading ? 0.25 : 0
+                
+                bag += Signal(after: labelDelay).animated(style: AnimationStyle.easeOut(duration: 0.25)) { _ in
+                    setLabelAlpha()
+                }
+                
+                bag += Signal(after: layoutDelay).animated(style: AnimationStyle.easeOut(duration: 0.25), animations: { _ in
```
Comment (Iamsamwhoami) [inline Src/Generic/Viewables/LoadableButton.swift]: I think the animations label can be removed.

like so:
```
.animated(style: AnimationStyle.easeOut(duration: 0.25)) { _ in }
```

File: Src/Generic/Viewables/LoadableButton.swift [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                     make.width.equalTo(buttonView.intrinsicContentSize.width + self.button.type.extraWidthOffset())
+                }
+            })
+            
+            func setLabelAlpha() {
+                if isLoading {
+                    buttonView.titleLabel?.alpha = 0
+                } else {
+                    buttonView.titleLabel?.alpha = 1
+                }
```
Comment (gustafguner) [inline Src/Generic/Viewables/LoadableButton.swift]: Consider making this a one-liner: `buttonView.titleLabel?.alpha = isLoading ? 0 : 1`

File: Src/Generic/Viewables/LoadableButton.swift [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            }
+            
+            func setSpinnerAlpha() {
+                if isLoading {
+                    spinner.alpha = 1
+                } else {
+                    spinner.alpha = 0
+                }
```
Comment (gustafguner) [inline Src/Generic/Viewables/LoadableButton.swift]: Here too! ✋

**Pull request merged**